src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1828,7 +1828,7 @@ impl Cage {
                     panic!("Somehow a normal file with an fd was truncated but there was no file object in rustposix?");
                 } else {
                     let sysfilename = format!("{}{}", FILEDATAPREFIX, inodenum);
-                    tempbind = interface::openfile(sysfilename, true).unwrap();
+                    tempbind = interface::openfile(sysfilename, filesize).unwrap(); // create new file
```

Comment:
Should this be length or ulength?

Reply:
this is just creating a new blank file, the actual truncating is done further down.

Reply:
If it's creating a blank file, should filesize be 0?  Why does it read from the inode on L1813?

Reply:
Right I said that wrong. It's opening a file that we know is of length filesize from the inode. The truncating based on length/ulength is done further down.